src/dialectic/core.py

```
@@ -390,55 +704,172 @@ async def answer(self, query: str) -> str:
+                max_tokens=search_max_tokens,
+                tools=tools,
+                tool_choice=level_settings.TOOL_CHOICE,
+                tool_executor=tool_executor,
+                max_tool_iterations=level_settings.MAX_TOOL_ITERATIONS,
+                messages=self.messages,
+                track_name="Dialectic Search",
+                thinking_budget_tokens=level_settings.THINKING_BUDGET_TOKENS,
+                max_input_tokens=settings.DIALECTIC.MAX_INPUT_TOKENS,
+                trace_name="dialectic_search",
+            )
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)